### PR TITLE
[tests-only][full-ci]send X-Request-Id in header of every request

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR includes the use of `X-Request-Id` header in each API requests. The `X-Request-Id` header consists the line reference of the scenario under test. If any of the test fails then oCIS server will log the error which contains `request-id` which represents the line reference of the failed scenario. For example, when any of the test fails oCIS server will provide the log as:

`
{"level":"error","service":"ocdav","name":"com.owncloud.web.ocdav","traceid":"00000000000000000000000000000000","request-id":"apiContract/propfind.feature:16-18","spaceid":"a428260f-6f47-4e9d-a939-6236977c3f8a$bbac7138-4375-4c27-8241-7ebe1e5af89d","path":"/","status":{"code":15,"message":"error getting upload id: Decomposedfs: error walking path: Decomposedfs: error walking path: decomposedfs: Wrap: readlink error: readlink /home/prajwol/.ocis/storage/users/spaces/bb/ac7138-4375-4c27-8241-7ebe1e5af89d/nodes/bb/ac/71/38/-4375-4c27-8241-7ebe1e5af89d: invalid argument","trace":"00000000000000000000000000000000"},"code":500,"time":"2023-09-05T15:14:31.495096701+05:45","message":"Internal Server Error"}
`

The server log consists of `"request-id":"apiContract/propfind.feature:16-18"`, which indicates the `apiContract` test suite, `propfind` feature, the scenario at line `16` and the test step at line `18` has failed. Using this server information tracking of the failed scenario and its step will be easier.

## Related Issue

- https://github.com/owncloud/ocis/issues/7022

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by saw-jan** [comment]
Please, add the example log in the PR description from the oCIS server logs which contain the requiest-id defined by the tests

File: tests/acceptance/features/apiContract/copy.feature [hunk 1]
```diff
@@ -19,7 +19,7 @@ Feature: Copy test
     When user "Alice" copies file "testfile.txt" from space "new-space" to "/new/testfile.txt" inside space "new-space" using the WebDAV API
     Then the HTTP status code should be "201"
     And the following headers should match these regular expressions
-      | Oc-Fileid                   | /^[a-f0-9!\$\-]{110}$/ |
-      | Access-Control-Allow-Origin | /^[*]{1}$/             |
-      | X-Request-Id                | /^[a-f0-9!\-]{36}$/    |
+      | Oc-Fileid                   | /^[a-f0-9!\$\-]{110}$/    |
+      | Access-Control-Allow-Origin | /^[*]{1}$/                |
+      | X-Request-Id                | /^[^\/]+\/[^:]+:\d+-\d+$/ |
```
Comment (saw-jan) [inline tests/acceptance/features/apiContract/copy.feature:24]: if we want to be more strict to the scenario line pattern. Also, it matches likes like `apiContract/propfind.feature:20` if there will be
```suggestion
      | X-Request-Id                | /^[a-zA-Z]+\/[a-zA-Z]+.feature:\d+(-\d+)?$/ |
```

**Review by SagarGi — COMMENTED** [review]
Other LGTM :+1:

File: tests/acceptance/features/apiContract/copy.feature (hunk 1 above)
Comment (saw-jan) [reply tests/acceptance/features/apiContract/copy.feature:24]: UPDATED:
```diff
-^[a-zA-Z]+\/[a-zA-Z]+.feature:\d+(-\d+)?$
+^[a-zA-Z]+\/[a-zA-Z]+\.feature:\d+(-\d+)?$
```

**Review by SwikritiT — COMMENTED** [review]
LGTM

**Review by saw-jan — APPROVED** [review]

**Pull request merged**